gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -0,0 +1,143 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "gematria/datasets/bhive_importer.h"
+#include "gematria/datasets/find_accessed_addrs.h"
+#include "gematria/llvm/canonicalizer.h"
+#include "gematria/llvm/llvm_architecture_support.h"
+#include "gematria/utils/string.h"
+
+#define INITIAL_REG_VALUE 0x10000
+#define INITIAL_MEM_VALUE 2147483647
+#define LLVM_EXEGESIS_MEM_DEF_PREFIX "# LLVM-EXEGESIS-MEM-DEF "
+#define LLVM_EXEGESIS_MEM_MAP_PREFIX "# LLVM-EXEGESIS-MEM-MAP "
+#define LLVM_EXEGESIS_MEM_NAME_PREFIX "MEM"
+
+static unsigned int mem_counter = 0;
+static unsigned int file_counter = 0;
+
+ABSL_FLAG(std::string, bhive_csv, "",
+          "Filename of the input file containing code hex");
+ABSL_FLAG(
+    std::string, output_dir, "",
+    "Directory containing output files that can be executed by llvm-exegesis");
+ABSL_FLAG(std::string, exegesis_template,
+          "gematria/datasets/llvm-exegesis_warpper.S",
+          "Template file for llvm-exegesis");
+
+int main(int argc, char* argv[]) {
+  absl::ParseCommandLine(argc, argv);
+
+  std::string bhive_filename = absl::GetFlag(FLAGS_bhive_csv);
```

Comment:
This is a bit nitpicky, but
```
const std::string bhive_filename = absl::GetFlag(FLAGS_bhive_csv);
```

I'd generally prefer adding `const` wherever possible. It helps avoiding unintended modifications of data, and also helps with reasoning about what can change when and where.